pkgs/development/libraries/stargate-libcds/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation rec {
+  pname = "stargate-libcds";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "stargateaudio";
+    repo = "libcds";
+    rev = version;
+    sha256 = "sha256-THThEzS8gGdwn3h0EBttaX5ljZH9Ma2Rcg143+GIdU8=";
+  };
+
+  # Remove unecessary tests (valgrind, coverage)
+  patches = [ ./Makefile.patch ];
+
+  doCheck = true;
+
+  installPhase = ''
+    runHook preInstall
+    install -D libcds.so -t $out/lib/
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "C data structure library";
+    homepage = "https://github.com/stargateaudio/libcds";
+    maintainers = with maintainers; [ yuu ];
+    license = licenses.lgpl3Only;
+    platforms = platforms.linux;
```

Comment:
@figsoda @jonringer do you know why it's not building? in the both https://logs.nix.ci/?attempt_id=ad3f4fff-eb4d-48b3-9304-ce333e1a2883&key=nixos%2Fnixpkgs.145157 and https://logs.nix.ci/?attempt_id=975e7679-0c8b-410c-87bb-76de152e2b9b&key=nixos%2Fnixpkgs.145157 are stuck on Cannot nix-instantiate `libcds.passthru.tests' because:
error: attribute 'tests' in selection path 'libcds.passthru.tests' not found

Reply:
They don't seem to be stuck for me
Despite saying error, `error: attribute 'tests' in selection path 'libcds.passthru.tests' not found` is more like a warning that gets ignored

This is the error I'm seeing on aarch64-linux build, I'm not familiar enough with gcc to know what's going on
```
# Compile the shared library
gcc \
    -Wall -fPIC -O2 -mfpmath=sse -mssse3 -shared \
    src/bytestr.c src/fast-sll.c src/hashtable/hash.c src/hashtable/entry.c src/hashtable/bucket.c src/queue.c src/list.c src/dl-list.c src/hashtable.c src/objectstore.c -Iinclude -o libcds.so
gcc: error: unrecognized command line option '-mfpmath=sse'
gcc: error: unrecognized command line option '-mssse3'
make: *** [Makefile:151: shared] Error 1
```

x86_64-darwin failed with a different error ([github actions](https://github.com/figsoda/nixpkgs-test/runs/4154835581)):
```
build flags: SHELL=/nix/store/vm564n0g8k6pkiwqb31x1dw1y02h2mak-bash-5.1-p8/bin/bash
# Compile the shared library
clang \
    -Wall -fPIC -O2 -mfpmath=sse -mssse3 -shared \
    src/objectstore.c src/bytestr.c src/list.c src/fast-sll.c src/dl-list.c src/queue.c src/hashtable.c src/hashtable/hash.c src/hashtable/bucket.c src/hashtable/entry.c -Iinclude -o libcds.so
src/list.c:2:10: fatal error: 'malloc.h' file not found
#include <malloc.h>
         ^~~~~~~~~~
1 error generated.
src/hashtable/bucket.c:1:10: fatal error: 'malloc.h' file not found
#include <malloc.h>
         ^~~~~~~~~~
1 error generated.
src/hashtable/entry.c:1:10: fatal error: 'malloc.h' file not found
#include <malloc.h>
         ^~~~~~~~~~
1 error generated.
make: *** [Makefile:151: shared] Error 1
```